BParkingNano/plugins/TriggeringMuonProducer.cc

```diff
@@ -0,0 +1,173 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/PatCandidates/interface/CompositeCandidate.h"
+
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/PatCandidates/interface/TriggerObjectStandAlone.h"
+#include "DataFormats/PatCandidates/interface/PackedTriggerPrescales.h"
+#include "DataFormats/PatCandidates/interface/TriggerPath.h"
+#include "DataFormats/PatCandidates/interface/TriggerEvent.h"
+#include "DataFormats/PatCandidates/interface/TriggerAlgorithm.h"
+
+#include <TLorentzVector.h>
+
+using namespace std;
+
+
+float MuonMass_ = 0.10565837;
+
+class TriggeringMuonProducer : public edm::EDProducer {
+    
+public:
+    
+    explicit TriggeringMuonProducer(const edm::ParameterSet &iConfig);
+    
+    ~TriggeringMuonProducer() override {};
+    
+    
+private:
+
+    virtual void produce(edm::Event&, const edm::EventSetup&);
+
+
+    edm::EDGetTokenT<std::vector<pat::Muon>> muonSrc_;
+    edm::EDGetTokenT<edm::TriggerResults> triggerBits_;
+    edm::EDGetTokenT<std::vector<pat::TriggerObjectStandAlone>> triggerObjects_;
+    edm::EDGetTokenT<pat::PackedTriggerPrescales> triggerPrescales_;
+    edm::EDGetTokenT<reco::VertexCollection> vertexSrc_;
+
+    bool debug;
+};
+
+
+TriggeringMuonProducer::TriggeringMuonProducer(const edm::ParameterSet &iConfig):
+  muonSrc_( consumes<std::vector<pat::Muon>> ( iConfig.getParameter<edm::InputTag>( "muonCollection" ) ) ),
+  triggerBits_(consumes<edm::TriggerResults>(iConfig.getParameter<edm::InputTag>("bits"))),
+  triggerObjects_(consumes<std::vector<pat::TriggerObjectStandAlone>>(iConfig.getParameter<edm::InputTag>("objects"))),
+  triggerPrescales_(consumes<pat::PackedTriggerPrescales>(iConfig.getParameter<edm::InputTag>("prescales"))),
+  vertexSrc_( consumes<reco::VertexCollection> ( iConfig.getParameter<edm::InputTag>( "vertexCollection" ) ) )
+{
+    produces<pat::CompositeCandidateCollection>();
```

Comment:
Why make `pat::CompositeCandidateCollection` instead of `pat::MuonCollection`?

Reply:
ok done